src/engine/routing_algorithms/map_matching.cpp

```diff
@@ -220,13 +221,27 @@ SubMatchingList mapMatching(SearchEngineData<Algorithm> &engine_working_data,
                         continue;
                     }
 
+                    PhantomNode prev_node = prev_unbroken_timestamps_list[s].phantom_node;
```

Comment:
A phantom node is quite big (~64 bytes), a copy in the hot loop of the code here will degrade performance significantly. I'm not even sure why you need this code: We don't allow u-turns by default, we only enable it when adjacent segments in the trace form a very sharp angle.